Pull in Upstream Changes

mcdonnnj opened the pull request:
# <!--- Provide a general summary of your changes in the Title above -->

## 🗣 Description
This PR pulls in the changes in cisagov/skeleton-generic#40 (and a bit more from before).
<!--- Describe your changes in detail -->

## 💭 Motivation and Context
Keep it up to date!
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## 🧪 Testing
`pre-commit` runs without issue.
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## 📷 Screenshots (if appropriate)

## 🚥 Types of Changes

<!--- What types of changes does your code introduce? -->
<!--- Put an `x` in all the boxes that apply: -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

<!--- Go over all the following points, and put an `x` in all the
boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask.
We're here to help! -->

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

felddy committed:
Autoupdate pre-commit hooks.  Add mypy.

felddy committed:
Merge pull request #35 from cisagov/improvement/static_type_checking

Autoupdate pre-commit hooks.  Add mypy.

jsf9k committed:
Add .mypy_cache to .gitignore

jsf9k committed:
Merge pull request #36 from cisagov/improvement/add-mypy-cache-to-gitignore

Add .mypy_cache to .gitignore

mcdonnnj committed:
Incorporate the Python version into keys for pip and pre-commit caches.
This should resolve the issue seen when the Python version changes before there
is an update to .pre-commit-config.yml which results in pre-commit pointing to
a non-existent Python installation.

mcdonnnj committed:
Merge pull request #37 from cisagov/add_python_version_to_cache_keys

Incorporate the Python version into keys for pip and pre-commit caches.

mcdonnnj committed:
Change the cache paths from hardcoded values in their appropriate blocks to environment variables declared before the job block.

mcdonnnj committed:
Merge pull request #38 from cisagov/change_cache_paths_to_env_variables

Change Cache Paths to Environment Variables

mcdonnnj committed:
Add a rule for markdownlint to allow multiple headers with the same name as long
as they are not nested in the same heading group.

mcdonnnj committed:
Merge pull request #39 from cisagov/add_mdl_rule_for_MD024

Add markdownlint Configuration Option for MD024

mcdonnnj committed:
Fixed broken hook id for pre-commit-terraform:
  terraform_validate_no_variables was changed to terraform_validate in the
  following commit:
  https://github.com/antonbabenko/pre-commit-terraform/commit/35e0356188b64a4c5af9a4e7200d936e514cba71.
Ran pre-commit autoupdate.

mcdonnnj committed:
Merge pull request #40 from cisagov/improvements/update_pre-commit

Update pre-commit Hooks and Fix Broken Hook ID

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/upstream_updates

dav3r reviewed (approved):
Fantastic!

mcdonnnj committed:
Add default provider information for the module.
Add Terraform initialization to the GitHub Actions workflow.

mcdonnnj committed:
Switch where the Terraform initialization takes place on the GitHub Actions
instance.

mcdonnnj committed:
We need to initialize the module and the example.

mcdonnnj committed:
Use my fork of the pre-commit-terraform hooks to allow the terraform_validate
hook to work with our directory tree.

mcdonnnj committed:
Adjust GitHub Actions workflow to eliminate superfluous 'cd ../..'.

felddy reviewed (approved):
🍪

jsf9k commented on main.tf:
```diff
@@ -7,6 +7,11 @@
 # The AWS account ID being used
 data "aws_caller_identity" "current" {}
 
+# Default provider information.
```
I suggest moving this provider block above the `aws_caller_identity`, or to a separate file called `providers.tf`.  No big deal if you disagree.

mcdonnnj replied:
Forgot to mention that I fixed this in 0176807 by moving it to `providers.tf`. Thanks for the feedback!

jsf9k reviewed (approved):
Very nice work, @mcdonnnj!  I am following antonbabenko/pre-commit-terraform#100.

I made one suggestion, which you can take or leave.

mcdonnnj committed:
More default provider information for the module out of main.tf and into the
providers.tf per excellent feedback.

mcdonnnj committed:
Adjust 'terraform init' commands in the GitHub Actions workflow to eliminate
changing directories.

mcdonnnj committed:
+Add caching of .terraform directory to preserve provisioners between runs.
+Add directive to find and initialize Terraform directories.
-Removed directive that searched for Terraform directories.

mcdonnnj committed:
Switch back from personal fork of pre-commit-terraform now that my PR has been
merged.

mcdonnnj committed:
Ran `pre-commit autoupdate` to get the latest version of the
`pre-commit-terraform` hook since the PR I submitted,
https://github.com/antonbabenko/pre-commit-terraform/pull/100,
was approved. This will fix issues with `skeleton-tf-module` related to multiple
directories with Terraform code.

mcdonnnj committed:
Explicitly install pre-commit hooks as its own step so it's clearer when the
failure is with setup rather than hooks running.

mcdonnnj committed:
Rename action to accurately reflect that we are only linting, not building.
Adjust description for pre-commit hook setup to better convey what is being
done.

mcdonnnj committed:
Sort .gitignore entries.

mcdonnnj committed:
Merge pull request #41 from cisagov/bugfix/update_pre-commit_for_pre-commit-terraform_hook

Update pre-commit Hooks to Include Bugfix

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/upstream_updates

dav3r reviewed (approved):
It's still all good from me. 👍 Thanks for helping us get our Terraform house in order.

jsf9k reviewed (approved):
Still looks good to me!  Approved!
